src/Microsoft.FeatureManagement/Targeting/ContextualTargetingFilter.cs

```diff
@@ -61,6 +61,27 @@ public Task<bool> EvaluateAsync(FeatureFilterEvaluationContext context, ITargeti
                 throw new ArgumentException(message, paramName);
             }
 
+            if (settings.Audience.Exclude != null)
+            {
+                //
+                // Check if the user is being excluded directly
+                if (targetingContext.UserId != null &&
+                    settings.Audience.Exclude.Users != null &&
+                    settings.Audience.Exclude.Users.Any(user => targetingContext.UserId.Equals(user, ComparisonType)))
+                {
+                    return Task.FromResult(false);
+                }
+
+                //
+                // Check if the user is in a group that is being excluded
+                if (targetingContext.Groups != null &&
+                    settings.Audience.Exclude.Groups != null &&
+                    settings.Audience.Exclude.Groups.Intersect(targetingContext.Groups).Any())
```

Comment:
Intersect isn't necessary, we only need a single occurence.

Reply:
Agreed, but .Intersect is O(N) while .Any is O(N^2) but faster when it's smaller and uses no memory. 

This answer has a good comparison of the two- https://stackoverflow.com/questions/17153688/intersect-and-any-or-contains-and-any-which-is-more-efficient-to-find-at-least

If you feel stronger about using .Any here, we can do that.

Reply:
But I also don't feel to strongly. If you like Intersect keep it, although I'd put a comment about why to use it over Any.

Reply:
I think that's good enough reasoning. I'll go ahead and switch it to Any.